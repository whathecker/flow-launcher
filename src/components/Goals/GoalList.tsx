import React from "react";
import { FlatList, ListRenderItem } from "react-native";
import { Goal } from "./components";
import { Goal as GoalEntity } from "../../types/core/entity";
import { View } from "../Themed";

type GoalListProps = {
  goals: GoalEntity[];
};

type GoalItemColor = "#E9695B" | "#84C5E8" | "#1C88BA" | "#F0B541";

function assignColor(index: number): GoalItemColor {
  if (_isFirstColor(index)) return "#E9695B";

  let color;
  switch (_findColorToRender(index)) {
    case 1:
      color = "#84C5E8";
      break;
    case 2:
      color = "#1C88BA";
      break;
    case 3:
      color = "#F0B541";
      break;
    default:
      color = "#E9695B";
      break;
  }

  return color as GoalItemColor;
}

function _isFirstColor(index: number): boolean {
  if (index === 0 || index % 4 === 0) return true;

  return false;
}

function _findColorToRender(index: number): number {
  return index % 4;
}

const GoalList: React.FC<GoalListProps> = ({ goals }: GoalListProps) => {
  const renderGoal: ListRenderItem<GoalEntity> = ({ item, index }) => {
    const backgroundColor = assignColor(index);
    return <Goal goal={item} backgroundColor={backgroundColor} />;
  };

  return (
    <View
      style={{
        width: "85%",
      }}
    >
      <FlatList
        data={goals}
        renderItem={renderGoal}
        keyExtractor={(item) => item._id as string}
      />
    </View>
  );
};

export default GoalList;
