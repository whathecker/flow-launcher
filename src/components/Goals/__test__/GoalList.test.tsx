import * as React from "react";
import renderer from "react-test-renderer";
import GoalList from "../GoalList";

describe("Test GoalList component", () => {
  it("Should renders GoalList correctly", () => {
    const tree = renderer.create(<GoalList />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
