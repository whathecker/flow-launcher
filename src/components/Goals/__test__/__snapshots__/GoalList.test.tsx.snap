// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GoalList component Should renders GoalList correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "width": "85%",
      },
    ]
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "_id": "sjdiodfi103jsldicsd",
          "motivation": "this is my motivation",
          "reminder": "daily",
          "status": "open",
          "tasks": Array [],
          "title": "dummy goal -1",
        },
        Object {
          "_id": "sjdiodfi103jsldicsd123123",
          "motivation": "this is my motivation",
          "reminder": "daily",
          "status": "open",
          "tasks": Array [],
          "title": "dummy goal -2",
        },
        Object {
          "_id": "sjdiodfi103jsldicsdasdasdd123123",
          "motivation": "this is my motivation",
          "reminder": "daily",
          "status": "open",
          "tasks": Array [],
          "title": "dummy goal -3",
        },
        Object {
          "_id": "sjdiodfi103jsldicsdasdasdd123121234",
          "motivation": "this is my motivation",
          "reminder": "daily",
          "status": "open",
          "tasks": Array [],
          "title": "dummy goal -4",
        },
        Object {
          "_id": "sjdiodfi103jsldicsdasdasdd12312sdfsd",
          "motivation": "this is my motivation",
          "reminder": "daily",
          "status": "open",
          "tasks": Array [],
          "title": "dummy goal -5",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#E9695B",
              "borderRadius": 10,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginBottom": 15,
              "marginLeft": "5%",
              "marginRight": "5%",
              "marginTop": 15,
              "minHeight": 200,
              "opacity": 1,
              "paddingBottom": 30,
              "paddingLeft": 40,
              "paddingRight": 60,
              "paddingTop": 30,
              "shadowColor": "#171717",
              "shadowOffset": Object {
                "height": 5,
                "width": 3,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-SemiBold",
                  "fontSize": 18,
                  "paddingBottom": 20,
                  "textAlign": "left",
                },
              ]
            }
          >
            dummy goal -1
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-Regular",
                  "fontSize": 16,
                },
              ]
            }
          >
            this is my motivation
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#84C5E8",
              "borderRadius": 10,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginBottom": 15,
              "marginLeft": "5%",
              "marginRight": "5%",
              "marginTop": 15,
              "minHeight": 200,
              "opacity": 1,
              "paddingBottom": 30,
              "paddingLeft": 40,
              "paddingRight": 60,
              "paddingTop": 30,
              "shadowColor": "#171717",
              "shadowOffset": Object {
                "height": 5,
                "width": 3,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-SemiBold",
                  "fontSize": 18,
                  "paddingBottom": 20,
                  "textAlign": "left",
                },
              ]
            }
          >
            dummy goal -2
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-Regular",
                  "fontSize": 16,
                },
              ]
            }
          >
            this is my motivation
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#1C88BA",
              "borderRadius": 10,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginBottom": 15,
              "marginLeft": "5%",
              "marginRight": "5%",
              "marginTop": 15,
              "minHeight": 200,
              "opacity": 1,
              "paddingBottom": 30,
              "paddingLeft": 40,
              "paddingRight": 60,
              "paddingTop": 30,
              "shadowColor": "#171717",
              "shadowOffset": Object {
                "height": 5,
                "width": 3,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-SemiBold",
                  "fontSize": 18,
                  "paddingBottom": 20,
                  "textAlign": "left",
                },
              ]
            }
          >
            dummy goal -3
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-Regular",
                  "fontSize": 16,
                },
              ]
            }
          >
            this is my motivation
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#F0B541",
              "borderRadius": 10,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginBottom": 15,
              "marginLeft": "5%",
              "marginRight": "5%",
              "marginTop": 15,
              "minHeight": 200,
              "opacity": 1,
              "paddingBottom": 30,
              "paddingLeft": 40,
              "paddingRight": 60,
              "paddingTop": 30,
              "shadowColor": "#171717",
              "shadowOffset": Object {
                "height": 5,
                "width": 3,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-SemiBold",
                  "fontSize": 18,
                  "paddingBottom": 20,
                  "textAlign": "left",
                },
              ]
            }
          >
            dummy goal -4
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-Regular",
                  "fontSize": 16,
                },
              ]
            }
          >
            this is my motivation
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-start",
              "backgroundColor": "#E9695B",
              "borderRadius": 10,
              "flexDirection": "column",
              "justifyContent": "center",
              "marginBottom": 15,
              "marginLeft": "5%",
              "marginRight": "5%",
              "marginTop": 15,
              "minHeight": 200,
              "opacity": 1,
              "paddingBottom": 30,
              "paddingLeft": 40,
              "paddingRight": 60,
              "paddingTop": 30,
              "shadowColor": "#171717",
              "shadowOffset": Object {
                "height": 5,
                "width": 3,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-SemiBold",
                  "fontSize": 18,
                  "paddingBottom": 20,
                  "textAlign": "left",
                },
              ]
            }
          >
            dummy goal -5
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Lexend-Regular",
                  "fontSize": 16,
                },
              ]
            }
          >
            this is my motivation
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
